12-watchtower-API.md

```diff
@@ -22,227 +22,320 @@ The scope of this bolt does not include:
+	* `wt_signature_algorithm` does not match any of the ones offered by the WatchTower
+	* `wt_signature` cannot be verified using `wt_public_key`.
+
+#### Receipt Format 
+
+The server (WatchTower) MUST create the receipt as containing the following infiormation:
+
+	txlocator
+	start_block
+	end_block
+	dispute_delta
+	encrypted_blob
+	transaction_size
+	transaction_fee
+	cipher
+	customer_public_key
+	customer_signature
+	wt_public_key
+	
 
-## Range of values 
+#### Rationale
 
-- ```start_block```: (Absolute) Block number 
-- ```end_block```: (Absolute) Block number
-- ```dispute_delta```: (Relative) Block number
-- ```transaction_size```: Measured in Bytes (e.g. 200) 
-- ```transaction_fee```: Measured in sats (e.g. 2000)
-- ```cipher```: AESGCM256, CHACHA20
-- ```customer_signature_algorithm```: ECDSA, SCHNORR
+We assume the server has a well-known public key for the WatchTower. 
 
-### Rationale
+The receipt contains, mainly, the information provided by the user. The WatchTower will need to sign the receipt to provide evidence of agreement.
 
-We can group the data fields into logical groups. 
+The `customer_signature` is included in the receipt to link both the client request and the server response. Otherwise a client could sign random data and claim to have send it to the tower. In the same way, a signed receipt protects the user from false claims from the tower.
 
-* **Appointment information**: The appointment time is defined using the ```start_block``` and```end_block```. WatchTower will delete the job when the appointment has expired. We recommend only using block numbers as that is the natural clock for Bitcoin. 
-* **Explicit acknowledgement of transaction details**: The ```dispute_delta```, ```transaction_size```, and ```transaction_fee``` let the WatchTower confirm the transaction is "reasonable" and it can be accepted into the blockchain (especially if there is congestion in the future). 
-* **Encrypted transaction**: The ```cipher```,```encrypted_blob``` states how the WatchTower can later find the dispute transaction and decrypt the justice transaction. 
-* **Customer signature** The ```customer_public_key``` and ```customer_signature``` provides an explicit message about the job from the customer. 
 
-Generally, this standard is trying to allow a reputationally accountable watching service. The signed job from the customer provides an explicit acknowledgement of the transaction details that is important for the WatchTower to decide whether they can accept it. If the decrypted justice transaction does not satisfy the signed job (e.g. fee too low), then the WatchTower is not obliged to fulfil it. 
+#### Receipt serialization and signature
 
-The _explictiness_ of the signed job ensures there is a clear protocol trasncript between the customer and WatchTower. Given the blockchain and decrypted justice transaction, anyone can verify that the WatchTower could have satisified the job. 
+[FIXME: TBD]
```

Comment:
Does this need to be filled in?

Reply:
Not yet, just leaving it as a placeholder